src/drivers/driver_zephyr.c

```diff
@@ -1102,6 +1182,127 @@ static void wpa_drv_zep_deinit(void *priv)
 	os_free(if_ctx);
 }
 
+#ifdef CONFIG_WIFI_NM_HOSTAPD_AP
+static void wpa_drv_zep_event_acs_channel_selected(struct zep_drv_if_ctx *if_ctx, union wpa_event_data *event)
+{
+	if (if_ctx->hapd)
+		hostapd_event_wrapper(if_ctx->hapd, EVENT_ACS_CHANNEL_SELECTED, event);
+	else
+		wpa_supplicant_event_wrapper(if_ctx->supp_if_ctx, EVENT_ACS_CHANNEL_SELECTED, event);
+}
+
+static void *wpa_drv_zep_hapd_init(struct hostapd_data *hapd, struct wpa_init_params *params)
+{
+	struct zep_drv_if_ctx *if_ctx              = NULL;
+	const struct zep_wpa_supp_dev_ops *dev_ops = NULL;
+	const struct device *device                = NULL;
+	const struct net_linkaddr *link_addr       = NULL;
+	struct zep_hostapd_dev_callbk_fns callbk_fns;
+
+	device = net_if_get_device(net_if_get_wifi_uap());
+	if (!device) {
+		wpa_printf(MSG_ERROR, "%s: Interface %s not found", __func__, params->ifname);
+		goto out;
+	}
+
+	if_ctx = os_zalloc(sizeof(*if_ctx));
+	if (if_ctx == NULL) {
+		goto out;
+	}
+
+	if_ctx->hapd  = hapd;
+	if_ctx->is_ap = 1;
+
+	if_ctx->dev_ctx = device;
+	if_ctx->drv_ctx = params->global_priv;
+	link_addr = net_if_get_link_addr(net_if_get_wifi_uap());
+	os_memcpy(params->own_addr, link_addr->addr, link_addr->len);
+
+	dev_ops = get_dev_ops(if_ctx->dev_ctx);
+	if (!dev_ops->hapd_init) {
+		wpa_printf(MSG_ERROR, "%s: No op registered for hapd init", __func__);
+		os_free(if_ctx);
+		if_ctx = NULL;
+		goto out;
+	}
+
+	os_memset(&callbk_fns, 0, sizeof(callbk_fns));
+
+	//callbk_fns.scan_done       = wpa_drv_zep_event_proc_scan_done;
+	callbk_fns.get_wiphy_res     = wpa_drv_zep_event_get_wiphy;
+	//callbk_fns.survey_res      = wpa_drv_zep_event_proc_survey_res;
+	callbk_fns.acs_channel_sel   = wpa_drv_zep_event_acs_channel_selected;
+	callbk_fns.mgmt_rx           = wpa_drv_zep_event_mgmt_rx;
+	//callbk_fns.eapol_rx        = wpa_drv_zep_event_proc_eapol_rx;
+	callbk_fns.mgmt_tx_status    = wpa_drv_zep_event_mgmt_tx_status;
+	callbk_fns.mac_changed       = wpa_drv_zep_event_mac_changed;
+	callbk_fns.chan_list_changed = wpa_drv_zep_event_chan_list_changed;
+	//callbk_fns.ecsa_complete   = wpa_drv_zep_event_ecsa_complete;
+	//callbk_fns.dfs_cac_started  = wpa_drv_zep_event_dfs_cac_started;
+	//callbk_fns.dfs_cac_finished = wpa_drv_zep_event_dfs_cac_finished;
```

Comment:
Why these commented lines, is this code still WIP and you expect to uncomment these soonish?

Reply:
Those comment out callback functions might be supported in the future. I already removed commented lines to keep code clean. Thanks!